p2p/host_test.go

```diff
@@ -93,24 +95,43 @@ func TestBlocklist(t *testing.T) {
 			errStr:  "gater disallows connection to peer",
 		},
 		{
-			desc:    "local IPv6 disallowed",
-			address: "/ip6/::/tcp/0",
-			errStr:  "gater disallows connection to peer",
+			desc:        "local IPv6 disallowed",
+			address:     "/ip6/::/tcp/0",
+			requireIPv6: true,
+			errStr:      "gater disallows connection to peer",
 		},
 		{
 			desc:           "local IPv4 allowed",
 			address:        "/ip4/127.0.0.1/tcp/0",
 			clearBlocklist: true,
 		},
 		{
-			desc:           "local IPv4 allowed",
+			desc:           "local IPv6 allowed",
 			address:        "/ip6/::/tcp/0",
+			requireIPv6:    true,
 			clearBlocklist: true,
 		},
 	}
 
+	ifs, err := net.InterfaceAddrs()
+	require.NoError(t, err)
+
+	ipv6Available := false
+
+	// check if at least one interface is IPv6
```

Comment:
Nit: an interface can have both IPv6 and IPv4 addresses, so speaking about an interface being IPv6 is not entirely correct

```suggestion
	// check if at least one interface has an IPv6 address
```